pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -66,6 +66,10 @@ const (
 	// machineDeploymentLabelKey defines the label key used to contains as value the MachineDeployment name
 	// which machine comes from.
 	machineDeploymentLabelKey = "md"
+	// httpSource defines the http source type for VM Disk Image.
+	httpSource = "http"
```

Comment:
I would suggest to introduce `imageSource` type.
```
type imageSource string

const (
   ...
   httpSource imageSource = "http"
   pvcSource imageSource = "pvc"
)
````

and later in the switch statement you would have

````
switch imageSource(osImageSource) ...
````